Libraries/process/Sources/DefaultLauncher.cpp

```diff
@@ -141,15 +147,22 @@ launch(Filesystem *filesystem, Context const *context)
         return ext::nullopt;
     }
 
-    if (!CloseHandle(process.hThread)) {
-        /* Error, but process started. Ignore. */
-    }
+    /* Wait until the spawned process finishes */
+    WaitForSingleObject(process.hProcess, INFINITE);
 
-    if (!CloseHandle(process.hProcess)) {
-        /* Error, but process started. Ignore. */
-    }
+    /* Get the exit code of the process */
+    DWORD status;
+    bool processFailed = GetExitCodeProcess(process.hProcess, &status);
+
+    /* Close process handles */
+    CloseHandle(process.hThread);
+    CloseHandle(process.hProcess);
 
-    return static_cast<int>(process.dwProcessId);
+    if (processFailed) {
```

Comment:
This is incorrect.

The return value of `GetExitCodeProcess()` is an integer that indicates whether or not `GetExitCodeProcess()` executed properly. It doesn't indicate anything about the child.

I think you want something like this:

```
if (GetExitCodeProcess(process.hProcess, &status) == 0) {
  // throw an exception, or assert, etc, because GetExciteCodeProcess failed.
}

[...]

return static_cast<int>(status);
```